fitbenchmarking/fitbenchmarking
# 680 Behaviour when incompatible options selected

#### Description of Work

Moves check for empty results outside of benchmark function, so that if all minimizers for a problem set fail then it can be ignored rather than causing an error.

If minimizer conflicts with analytic jacobian option then instead of an error, a warning is produced and option is ignored.

A function has been added to the base cost function classes to check that the selected minimizer is compatible with the selected cost function e.g if algorithm_type all has been selected with poisson cost function then the function will identify any ls minimizers that will then be ignored.

Finally when results are processed, if both the poisson cost function and local_min table have been selected, then the local min table option is ignored to avoid an error

Fixes [#680](https://github.com/fitbenchmarking/fitbenchmarking/issues/680)

#### Testing Instructions

1. Try running various options/problems to see if above changes work as intended
2. Do warning messages make sense and are they helpful to the user?

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
